qa/workunits/rbd/rbd-nbd.sh

```diff
@@ -408,5 +424,25 @@ if [ -n "${COOKIE}" ]; then
     test "${ANOTHER_COOKIE}" = "abc de"
     unmap_device ${DEV} ${PID}
 fi
+OUT=
+
+# test detach/attach with --snap-id
+SNAPID=`_sudo rbd snap ls ${POOL}/${IMAGE} --all | grep -w snap | awk '{print $1;}'`
+OUT=`_sudo rbd device --device-type nbd --options try-netlink,show-cookie map --snap-id=${SNAPID} ${POOL}/${IMAGE}`
+read DEV COOKIE <<< "${OUT}"
+get_pid ${POOL}
+_sudo rbd device detach ${POOL}/${IMAGE} --snap-id=${SNAPID} --device-type nbd
+expect_false get_pid ${POOL}
+expect_false _sudo rbd device attach --device ${DEV} --snap-id=${SNAPID} ${POOL}/${IMAGE} --device-type nbd
+if [ -n "${COOKIE}" ]; then
+    _sudo rbd device attach --device ${DEV} --cookie ${COOKIE} --snap-id=${SNAPID} ${POOL}/${IMAGE} --device-type nbd
+else
+    _sudo rbd device attach --device ${DEV} --snap-id=${SNAPID} ${POOL}/${IMAGE} --device-type nbd --force
+fi
+get_pid ${POOL}
+_sudo rbd device detach ${DEV} --device-type nbd
```

Comment:
Did you mean to detach without specifying `SNAPID` here?

Reply:
Yep

Reply:
... because it's a detach by device, not by image name, I see.  Thanks!